backend/src/resolvers.js

```diff
@@ -0,0 +1,56 @@
+export const resolvers = {
+  Query: {
+    // listallposts
+    posts: async (parent, args, context) => {
+      return context.dataSources.postdb.getAll()
+    },
+    // listallusers
+    users: async (parent, args, context) => {
+      return context.dataSources.userdb.getAll()
+    }
+  },
+  Mutation: {
```

Comment:
General Comment for the mutations: 
early of the needed parameters might good here.
And it might also be good and easy to understand if the checks and the logic can be removed from the data sources and put in here so that the data sources only read and write data.